agent/agent.go

```diff
@@ -855,10 +865,11 @@ func (a *Agent) listenHTTP() ([]apiServer, error) {
 			}
 			a.configReloaders = append(a.configReloaders, srv.ReloadConfig)
 			a.httpHandlers = srv
+
 			httpServer := &http.Server{
 				Addr:           l.Addr().String(),
 				TLSConfig:      tlscfg,
-				Handler:        srv.handler(a.config.EnableDebug),
+				Handler:        otelhttp.NewHandler(srv.handler(a.config.EnableDebug), "Consul HTTP Request"),
```

Comment:
Open Source tracing for HTTP Handlers. Creates a span named `Consul HTTP Request` for every request